File: src/org/baksia/rustycage/preferences/PreferenceInitializer.java
package org.baksia.rustycage.preferences;

import org.baksia.rustycage.RustPlugin;
import org.eclipse.core.runtime.preferences.AbstractPreferenceInitializer;
import org.eclipse.jface.preference.IPreferenceStore;

public class PreferenceInitializer extends AbstractPreferenceInitializer {

    public void initializeDefaultPreferences() {
        IPreferenceStore store = RustPlugin.getDefault().getPreferenceStore();
        store.setDefault(PreferenceConstants.P_BOOLEAN, true);
        store.setDefault(PreferenceConstants.P_CHOICE, "choice2");
        store.setDefault(PreferenceConstants.P_STRING,
                "Default value");

        store.setDefault(PreferenceConstants.RUST_C, "/usr/local/bin/");
        store.setDefault(PreferenceConstants.P_PATH, "");
    }

}

File: src/org/baksia/rustycage/preferences/PreferenceInitializer.scala
package org.baksia.rustycage.preferences

import org.eclipse.core.runtime.preferences.AbstractPreferenceInitializer
import org.baksia.rustycage.RustPlugin

class PreferenceInitializer extends AbstractPreferenceInitializer {
  protected def initializeDefaultPreferences() {
    import PreferenceConstants._
    val store = RustPlugin.getDefault().getPreferenceStore
    store.setDefault(P_BOOLEAN, true)
    store.setDefault(P_CHOICE, "choice2")
    store.setDefault(P_STRING, "Default value")

    store.setDefault(RUST_C, "/usr/local/bin/")
    store.setDefault(P_PATH, "")
  }
}
